src/nvim/option.c

```diff
@@ -1490,7 +1490,7 @@ do_set (
               new_value_alloced = true;
               if (newval == NULL) {
                 newval = empty_option;
-              } else if (!(options[opt_idx].flags | P_NO_DEF_EXP)) {
+              } else if (!(options[opt_idx].flags & P_NO_DEF_EXP)) {
```

Comment:
Pretty sure, since (1) this is how flags are generally used, (2) this is how flags are used in the rest of the file.